platform/mbed_error.h

```diff
@@ -60,7 +67,17 @@
 extern "C" {
 #endif
 
-void error(const char* format, ...);
+
+static inline void error(const char *format, ...) {
+#ifndef NDEBUG
+    va_list arg;
+    va_start(arg, format);
+    mbed_error_vfprintf(format, arg);
+    va_end(arg);
+#endif
+    while (1) {}
```

Comment:
Let me know if there is something better to do here, should I just call mbed_die?

Even with the change in #2741, the current definition of error is still preventing the format strings/arguments from being garbage collected by this compiler.